bin/generate_data.js

```diff
@@ -4,21 +4,27 @@ const dataSource = require('../minecraft-data/data/dataPaths')
 const fs = require('fs')
 const path = require('path')
 
-const data = 'module.exports =\n{\n' + Object
-  .keys(dataSource)
-  .map(k1 =>
-    "  '" + k1 + "': {\n" + Object
-      .keys(dataSource[k1])
-      .map(k2 =>
-        "    '" + k2 + "': {" + '\n' + Object
-          .keys(dataSource[k1][k2])
-          .map(k3 => "      '" + k3 + "': require('./minecraft-data/data/" + dataSource[k1][k2][k3] + '/' + k3 + ".json')")
-          .join(',\n') +
-      '\n    }'
-      )
-      .join(',\n') +
-    '\n  }'
-  )
-  .join(',\n') + '\n}\n'
+function generate () {
+  let str = 'module.exports =\n{\n'
```

Comment:
you really should try to avoid refactoring code for no reason when you want to do one change
it makes things hard to review
here going from functional to imperative is not needed

Reply:
I tried to improve readability a bit - still need refactoring